### PR TITLE
Refactor NullRouteImpl out of async client

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
Commit Message: Refactor NullRouteImpl out of async client. 
Additional Description:
Risk Level: low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/common/http/BUILD [hunk 1]
```diff
@@ -12,8 +12,14 @@ envoy_package()
 
 envoy_cc_library(
     name = "async_client_lib",
-    srcs = ["async_client_impl.cc"],
-    hdrs = ["async_client_impl.h"],
+    srcs = [
+        "async_client_impl.cc",
+        "null_route_impl.cc",
```
Comment (yanavlasov) [inline source/common/http/BUILD:17]: Please create a separate target for the null_route_impl.cc library, since it is canonical approach with bazel.

**Review by yanavlasov — CHANGES_REQUESTED** [review]
/wait

File: source/common/http/BUILD (hunk 1 above)
Comment (vikaschoudhary16) [reply source/common/http/BUILD]: done, thanks!

File: source/extensions/early_data/default_early_data_policy.h
Comment (yanavlasov) [inline source/extensions/early_data/default_early_data_policy.h:1]: Is this file still needed? Looks like it can be removed.

**Review by yanavlasov — CHANGES_REQUESTED** [review]
/wait

File: source/extensions/early_data/default_early_data_policy.h
Comment (vikaschoudhary16) [reply source/extensions/early_data/default_early_data_policy.h:1]: it is still being used in NullRoute here, https://github.com/envoyproxy/envoy/pull/28638/files/3bfa74da7c8078d4b6915c263fd17aa5b46a3608#diff-d7f483e1142b03905f8668f2f86fcc9c68c60a7b1ce04c2360c902bd6192bf14R207-R208 . Sorry I could not get how this can be removed.

**Comment by ohadvano** [comment]
Ping @yanavlasov

**Comment by vikaschoudhary16** [comment]
Ping @yanavlasov

File: source/extensions/early_data/default_early_data_policy.h
Comment (yanavlasov) [reply source/extensions/early_data/default_early_data_policy.h:1]: I see. All it does is just including the "envoy/extensions/early_data/v3/default_early_data_policy.pb.h". Can you just include the "envoy/extensions/early_data/v3/default_early_data_policy.pb.h" where the default_early_data_policy.h is included and then get rid of the default_early_data_policy.h ?

**Comment by yanavlasov** [comment]
LGTM otherwise

/wait

File: source/extensions/early_data/default_early_data_policy.h
Comment (vikaschoudhary16) [reply source/extensions/early_data/default_early_data_policy.h:1]:
```
std::unique_ptr<Router::EarlyDataPolicy> early_data_policy_{
      new envoy::extensions::early_data::v3::DefaultEarlyDataPolicy()};
```
`EarlyDataPolicy` is an abstract class which is implemented by [DefaultEarlyDataPolicy](https://github.com/envoyproxy/envoy/blob/main/source/extensions/early_data/default_early_data_policy.h#L14). default_early_data_policy.pb.h cannot give us an instance of Router::EarlyDataPolicy. 
Also I want to avoid making any functional changes in this PR.

Comment (yanavlasov) [reply source/extensions/early_data/default_early_data_policy.h:1]: Oh sorry, I have missed that there is code below and thought the file was empty.

**Review by yanavlasov — APPROVED** [review]
